gusto/equations/shallow_water_equations.py

```diff
@@ -165,13 +165,13 @@ def __init__(self, domain, parameters, fexpr=None, bexpr=None,
 
         if fexpr is not None:
             V = FunctionSpace(domain.mesh, 'CG', 1)
-            f = self.prescribed_fields('coriolis', V).interpolate(fexpr)
+            self.f = self.prescribed_fields('coriolis', V).interpolate(fexpr)
```

Comment:
You don't need to put `f` in `self` - you can access it later through `self.prescribed_fields('coriolis')`